xla/service/gpu/cublas_padding_requirements.cc

```diff
@@ -27,20 +27,27 @@ namespace gpu {
 namespace {
 
 bool DimensionRequiresPadding(const int64_t size, const PrimitiveType data_type,
-                              const se::CudaComputeCapability cc) {
-  for (const CublasPaddingRequirement& requirement :
-       CublasPaddingRequirements) {
-    if (cc.IsAtLeast(requirement.min_compute_capability) &&
-        data_type == requirement.data_type &&
-        size % requirement.multiple_of != 0) {
-      return true;
+                              const se::GpuComputeCapability cc) {
+
+  if (auto pcc = std::get_if<se::CudaComputeCapability>(&cc); pcc) {                                
```

Comment:
`auto* pcc` to make it clear that it's a pointer

even better use a std::visit here

```
struct RequiresPaddingVisitor {
  bool operator(se::CudaComputeCapability cc) {}
}

return std::visit(RequiresPaddingVisitor{}, cc);
```

Reply:
yes, std::visit is a good idea, indeed !